source/link/linker.cpp

```diff
@@ -16,1183 +16,277 @@
-    case spv::Op::OpEndStreamPrimitive:
-      transform_id(binary, i + 1u);
-      break;
-    case spv::Op::OpControlBarrier:
-    case spv::Op::OpMemoryBarrier:
-    case spv::Op::OpGroupAsyncCopy:
-    case spv::Op::OpGroupWaitEvents:
-    case spv::Op::OpGroupAll:
-    case spv::Op::OpGroupAny:
-      TransformConsecutiveIDs(1u, insn_word_count);
-      break;
-    case spv::Op::OpGroupBroadcast:
-    case spv::Op::OpGroupIAdd:
-    case spv::Op::OpGroupUMin:
-    case spv::Op::OpGroupSMin:
-    case spv::Op::OpGroupUMax:
-    case spv::Op::OpGroupSMax:
-      TransformConsecutiveIDs(1u, insn_word_count);
-      RegisterVariable(i + 1u, i + 2u);
-      break;
-    case spv::Op::OpGroupFAdd:
-    case spv::Op::OpGroupFMin:
-    case spv::Op::OpGroupFMax:
-    case spv::Op::OpSubgroupBallotKHR:
-      TransformConsecutiveIDs(1u, insn_word_count);
-      break;
-    case spv::Op::OpSubgroupFirstInvocationKHR:
-    case spv::Op::OpEnqueueMarker:
-    case spv::Op::OpEnqueueKernel:
-    case spv::Op::OpGetKernelNDrangeSubGroupCount:
-    case spv::Op::OpGetKernelNDrangeMaxSubGroupSize:
-    case spv::Op::OpGetKernelWorkGroupSize:
-    case spv::Op::OpGetKernelPreferredWorkGroupSizeMultiple:
-      TransformConsecutiveIDs(1u, insn_word_count);
-      RegisterVariable(i + 1u, i + 2u);
-      break;
-    case spv::Op::OpRetainEvent:
-    case spv::Op::OpReleaseEvent:
-    case spv::Op::OpCreateUserEvent:
-    case spv::Op::OpIsValidEvent:
-    case spv::Op::OpSetUserEventStatus:
-    case spv::Op::OpCaptureEventProfilingInfo:
-    case spv::Op::OpGetDefaultQueue:
-    case spv::Op::OpBuildNDRange:
-      TransformConsecutiveIDs(1u, insn_word_count);
-      break;
-    case spv::Op::OpReadPipe:
-    case spv::Op::OpWritePipe:
-    case spv::Op::OpReservedReadPipe:
-    case spv::Op::OpReservedWritePipe:
-    case spv::Op::OpGetNumPipePackets:
-    case spv::Op::OpGetMaxPipePackets:
-      TransformConsecutiveIDs(1u, insn_word_count);
-      RegisterVariable(i + 1u, i + 2u);
-      break;
-    case spv::Op::OpReserveReadPipePackets:
-    case spv::Op::OpReserveWritePipePackets:
-    case spv::Op::OpCommitReadPipe:
-    case spv::Op::OpCommitWritePipe:
-    case spv::Op::OpIsValidReserveId:
-    case spv::Op::OpGroupReserveReadPipePackets:
-    case spv::Op::OpGroupReserveWritePipePackets:
-    case spv::Op::OpGroupCommitReadPipe:
-    case spv::Op::OpGroupCommitWritePipe:
-      TransformConsecutiveIDs(1u, insn_word_count);
-      break;
-    case spv::Op::Max:
-      return libspirv::DiagnosticStream(position, context.consumer,
-                                        SPV_ERROR_INVALID_DATA)
-             << "Invalid opcode: OpMax.";
-    default:
-      return libspirv::DiagnosticStream(position, context.consumer,
-                                        SPV_ERROR_INVALID_DATA)
-             << "Unsupported opcode: " << static_cast<uint32_t>(opcode) << ".";
-    }
-
-    position.index = i += insn_word_count;
-  }
+} // anonymous namespace
```

Comment:
Same as above. I was very surprised initially to read your comment about deleting it, as I was 100% sure I had already done that.